src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -188,32 +180,32 @@ public Void call() throws Exception {
     @Override
     public String createDeviceMember(final String name, String ipAddress, String vserverIpAddress, String contactIpAddress,
             String gateway, String prefixLength) throws Exception {
-        DeviceEntity device = new DeviceEntity();
         DeviceMemberEntity deviceMember = new DeviceMemberEntity();
-        deviceMember.setParent(device);
-        device.setId(null);
+        updateMember(deviceMember, vserverIpAddress, contactIpAddress, ipAddress, gateway, prefixLength);
         deviceMember.setName(name);
         return createDeviceMember(deviceMember, null, this.vs.getId());
     }
 
-    public String createDeviceMember(DeviceMemberEntity entity, Long parentId, Long vsId) throws Exception {
+    public String createDeviceMember(DeviceMemberEntity entity, Long deviceId, Long vsId) throws Exception {
         return this.txControl.required(new Callable<DeviceMemberEntity>() {
             @Override
             public DeviceMemberEntity call() throws Exception {
                 CriteriaBuilder criteriaBuilder = IsmDeviceApi.this.em.getCriteriaBuilder();
                 CriteriaQuery<DeviceEntity> query = criteriaBuilder.createQuery(DeviceEntity.class);
                 Root<DeviceEntity> r = query.from(DeviceEntity.class);
```

Comment:
to find the device entity use the existing method or use a consistent way to query the DB.

I see places where we are using criteria queries in some places and in other using em.find() when querying by primary keys....

Reply:
Ok, I have made it consistent wherever primary key based search is used. I am now using find method where the above condition is met iso of query based.